### PR TITLE
Jetpack Section: Scan Status

### PR DESCRIPTION
**Pull request by emilylaguna**
Project: https://github.com/wordpress-mobile/WordPress-iOS/issues/15190

This adds the initial logic and view states for the scan status view.

No buttons work, and the scanning view doesn't display progress yet.

### Idle
| No Threats (okay) | Has Fixable Threats | Has Unfixable Threats | Single Threat|
|:---:|:---:|:---:|:---:|
|no threats|multiple fixable threats|multiple unfixable threats|single unfixable threat|

### Scanning
| Preparing | Scanning |
|:---:|:---:|
|preparing|scanning|

### To test:
1. Launch the app
2. Tap on the My Site tab
3. Tap on a site with Jetpack Scan enabled
4. Tap the Scan item
5. Note the different states of the view
6. On the web trigger a scan
7. Tap back and tap scan again to reload the view
8. Wait for the scan to start
9. Tap back and tap scan again to reload the view 
10. Repeat for each state

### PR submission checklist:

- [ ] I have considered adding unit tests where possible.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class JetpackScanCoordinator {
+    private let service: JetpackScanService
+    private let blog: Blog
+    private let view: JetpackScanView
+
+    private(set) var scan: JetpackScan?
+
+    init(blog: Blog, view: JetpackScanView,
+        service: JetpackScanService? = nil,
+        context: NSManagedObjectContext = ContextManager.sharedInstance().mainContext) {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanCoordinator.swift]: [Nit] 
Indentation alignment seems to be off, and the `view:` parameter would look better on its own line

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    // MARK: - Private: 
+    private func configureTableView() {
+        tableView.register(UINib(nibName: String(describing: JetpackScanStatusCell.self), bundle: nil),
+                           forCellReuseIdentifier: Constants.statusCellIdentifier)
+
+        tableView.register(UINib(nibName: String(describing: JetpackScanThreatCell.self), bundle: nil),
+                           forCellReuseIdentifier: Constants.threatCellIdentifier)
+
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift]: What do you think of conforming to the `NibReusable` protocol for the cells?

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/View Models/JetpackScanStatusViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    private enum StatusViewState {
+        case noThreats
+        case hasThreats
+        case hasFixableThreats
+        case preparingToScan
+        case scanning
+        case error
+    }
+
+    private static func viewState(for scan: JetpackScan) -> StatusViewState {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/View Models/JetpackScanStatusViewModel.swift:89]: Nice! makes it easy to follow how scan states map to view states ✨

**Review by momo-ozawa — APPROVED** [review]
@emilylaguna Works as described!
Added a couple of minor suggestions.

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift [hunk 4]
```diff
@@ -0,0 +1,121 @@
+import UIKit
+
+class JetpackScanViewController: UIViewController, JetpackScanView {
+    private let blog: Blog
+
+    // IBOutlets
+    @IBOutlet weak var tableView: UITableView!
+
+    //
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift]: Missing comment?

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanCoordinator.swift (hunk 1 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanCoordinator.swift]: oops, thanks for the catch! Fixed in: d0f84cd49040e785e200e11f360b836dbe926187

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift (hunk 2 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift]: Done: 67315da65b17a416648f682216a0c59f2e8cc54e

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift (hunk 4 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift]: Done: ff07461d4a22dd4050c97111519219e3cba3874b

**Pull request merged**